File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {number} p
     * 
     */
    // dfs(root, p, q) {
    //     if (!root) {return false}
    //     if (root.val)

    // }

    /**
     * @param {TreeNode} root
     * @param {TreeNode} p
     * @param {TreeNode} q
     * @return {TreeNode}
     */
    lowestCommonAncestor(root, p, q) {
       if (!root || !q || !p) {return null}

       if (Math.max(q.val, p.val) < root.val) {return this.lowestCommonAncestor(root.left, p, q)} 
       else if (Math.min(q.val, p.val) > root.val) {
        return this.lowestCommonAncestor(root.right, p, q)
       } else {return root}
    } 
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.go
func minSubArrayLen(target int, nums []int) int {
	res, sum := math.MaxInt32, 0
	left := 0 

	for right := 0; right < len(nums); right++ {
		sum+=nums[right]
		for right >= left && sum >= target {
			sum-=nums[left]
			res = min(res, right - left + 1)
			left++
		}
	} 

	if res == math.MaxInt32 {res = 0}
	return res
}
